lib/iris/tests/integration/plot/test_vector_plots.py

```diff
@@ -167,6 +167,26 @@ def test_fail_unsupported_coord_system(self):
             self.plot('2d_rotated', u_cube, v_cube,
                       coords=('longitude', 'latitude'))
 
+    def test_circular_longitude(self):
+        # Test circular longitude does not cause a crash.
+        res = 5
+        lat = DimCoord(np.arange(-90, 91, res), 'latitude',
+                       units='degrees_north')
+        lon = DimCoord(np.arange(0, 360, res), 'longitude',
+                       units='degrees_east')
+        nlat = len(lat.points)
+        nlon = len(lon.points)
+        u_arr = np.ones((nlat, nlon))
+        v_arr = np.ones((nlat, nlon))
+        u_cube = Cube(u_arr, dim_coords_and_dims=[(lat, 0), (lon, 1)],
+                      standard_name='eastward_wind')
+        v_cube = Cube(v_arr, dim_coords_and_dims=[(lat, 0), (lon, 1)],
+                      standard_name='northward_wind')
+        u_cube.coord('longitude').circular = True
```

Comment:
Could you also set `circular=True` for v_cube?
I know we don't explicitly do a compatibility check for the u_cube and v_cube as it is still a [TODO](https://github.com/SciTools/iris/blob/master/lib/iris/plot.py#L1301) but once that is fixed, it would throw up an error about the longitude coordinates being different on u_cube and v_cube.

Also, you could do it when you initialise the `lon`, i.e.:
```
lon = DimCoord(np.arange(0, 360, res), 'longitude',
                       units='degrees_east', circular=True)
```
As that would apply to both u_cube and v_cube